src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1914,11 +1922,14 @@ DependencyPropertyValuePrecedences newPrecedence
 			propertyDetails.RaisePropertyChanged(actualInstanceAlias, eventArgs);
 
 			// Raise the property change for generic handlers
-			var currentCallbacks = _genericCallbacks.Data;
-			for (var callbackIndex = 0; callbackIndex < currentCallbacks.Length; callbackIndex++)
+			var currentCallbacks = _genericCallbacks;
```

Comment:
The reason behind the immutable lists is that callbacks may be changed while being enumerated. Did you see anything specific around the of the immutable list?

Reply:
Judging by the code it does, it appears expensive. I'll see if there is an alternative that performs well and doesn't have issues when the collection is modified while being enumerated.